File: javascript/leetcode/21. merge two sorted lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
  const finalList = new ListNode(0);
  let currentList = finalList;

  while (l1 !== null && l2 !== null) {
    if (l1.val < l2.val) {
      currentList.next = l1;
      l1 = l1.next;
    } else {
      currentList.next = l2;
      l2 = l2.next;
    }

    currentList = currentList.next;
  }

  if (l1 !== null) {
    currentList.next = l1;
    l1 = l1.next;
  }

  if (l2 !== null) {
    currentList.next = l2;
    l2 = l2.next;
  }

  return finalList.next;
};

File: go/leetcode/21. merge two sorted lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
    finalList := new(ListNode)
    currentList := finalList
    
    
    for l1 != nil && l2 != nil {
        if l1.Val < l2.Val {
            currentList.Next = l1
            l1 = l1.Next            
        } else {
            currentList.Next = l2
            l2 = l2.Next
        }
        currentList = currentList.Next
    } 
    if l1 != nil {
        currentList.Next = l1
        l1 = l1.Next 
    }
    if l2 != nil {
        currentList.Next = l2
        l2 = l2.Next
    }
     
    return finalList.Next
}